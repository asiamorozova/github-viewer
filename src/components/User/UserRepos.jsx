import React from 'react';
import PropTypes from 'prop-types';
import UserRepo from './UserRepo';

const UserRepos = ({ repos }) => {
  const repoData = repos.map(repo => <UserRepo key={repo.id} {...repo} />);
  return (
    <ul>
      {repoData}
    </ul>
  );
};

UserRepos.propTypes = {
  repos: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    html_url: PropTypes.string.isRequired
  }))
};

export default UserRepos;
