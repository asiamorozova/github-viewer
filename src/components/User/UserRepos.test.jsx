import React from 'react';
import { shallow } from 'enzyme';
import UserRepos from './UserRepos';

describe('Repos component', () => {
  it('renders Repos', () => {
    const repos = [
      {
        id: 233723653,
        name: 'random-twitter',
        html_url: 'https://github.com/asiamorozova/random-twitter'
      },
      {
        id: 236626883,
        name: 'cursed-banana',
        html_url: 'https://github.com/asiamorozova/cursed-banana'
      }
    ];
    const wrapper = shallow(<UserRepos repos={repos}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
