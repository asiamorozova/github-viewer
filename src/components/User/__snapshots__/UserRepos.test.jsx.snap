// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos component renders Repos 1`] = `
<ul>
  <UserRepo
    html_url="https://github.com/asiamorozova/random-twitter"
    id={233723653}
    key="233723653"
    name="random-twitter"
  />
  <UserRepo
    html_url="https://github.com/asiamorozova/cursed-banana"
    id={236626883}
    key="236626883"
    name="cursed-banana"
  />
</ul>
`;
